### PR TITLE
[558032] Add Title Block content sanitization

### PR DESCRIPTION
**Pull request by sandupostaru**
- Sanitize title block content in order to avoid adding invalid semantic
references to external resources in the .aird

- Fix Title Block Cell content display for Collections of primitive data
-- Only the last element in the collection was displayed

Bug: 558032
Change-Id: Ifb5e9efa69479ebe8b58a2b34ed6fccf65387ea0
Signed-off-by: Sandu Postaru <sandu.postaru@thalesgroup.com>

### CODE REVIEW

**Comment by sandupostaru** [comment]
Here is the current status:
TitleBlockCellContentSanitization

File: core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/TitleBlockHelper.java [hunk 1]
```diff
@@ -661,4 +643,98 @@ public static String getTitleBlockName(String type) {
...
+
+  private static boolean isValidResultItem(Object item) {
+    if (item instanceof EObject) {
+      EObject eObject = (EObject) item;
+
+      if (eObject instanceof Element || eObject instanceof DRepresentationDescriptor) {
+        return true;
+      }
+
+      EPackage ePackage = eObject.eClass().getEPackage();
```
Comment (minhtutonthat) [inline core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/TitleBlockHelper.java:694]: Could you extract this code into a separate method (e.g. isViewpointElement)?

File: core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/TitleBlockHelper.java [hunk 2]
```diff
@@ -661,4 +643,98 @@ public static String getTitleBlockName(String type) {
...
+    }
+
+    return resultItem != null ? resultItem.toString() : "";
+  }
+
+  private static Object sanitizeResultItems(Collection<?> originalResult) {
+    return originalResult.stream().map(TitleBlockHelper::sanitizeResultItem).collect(Collectors.toList());
+  }
+
+  private static Set<EPackage> getViewpointPackages() {
```
Comment (minhtutonthat) [inline core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/TitleBlockHelper.java:719]: Could you move this to a separated helper class?

File: core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/TitleBlockHelper.java (hunk 1 above)
Comment (sandupostaru) [reply core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/TitleBlockHelper.java]: Done

File: core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/TitleBlockHelper.java (hunk 2 above)
Comment (sandupostaru) [reply core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/TitleBlockHelper.java]: Done

**Pull request merged**